pehlaschoollibrary/proj.android-studio/app/src/com/maq/pehlaschool/library/MainActivity.java

```diff
@@ -109,7 +107,14 @@ protected void onCreate(Bundle savedInstanceState) {
 
         Log.d(TAG, "onCreate()");
         Util.hideSystemUI(this);
-
+        Intent localeIntent = getIntent();
+        Bundle extras = localeIntent.getExtras();
+        if (extras != null && extras.getString("locale") != null && !extras.getString("locale").isEmpty()) {
+            locale = extras.getString("locale").toLowerCase();
+        } else {
+            // set the default value of the variable on successive calls
+            locale = "english";
+        }
         appLanguage = "en-us";
```

Comment:
Add comment for this block of code. If this block is repeated somewhere else then create a method for it.

Reply:
Added comment and used function getLocalefromIntent() to get the locale.